### DISCUSSION TITLE
Auto suspend a Kustomization once reconciled?

### DISCUSSION THREAD

**Discussion by ntx-ben** [fluxcd/flux2#2689] [General] [answered]
I'm looking into how I could suspend automatically a `Kustomization` once it has been successfully reconciled (i.e. a `Kustomization` would reconcile a Kubernetes `Job`, and I want that to happen only once).

I've looked into the `PostBuild` hook to set the `suspend `field, but I don't think I can use that. Any ideas?

Thanks

**Comment by cwyl02** [comment]
~https://fluxcd.io/docs/components/helm/api/#helm.toolkit.fluxcd.io/v2beta1.PostRenderer worth a shot~

**Reply by kingdonb** [reply]
The suggestion is, I think, to patch the Kustomization as though it is a "pre-existing resource" like this:

https://fluxcd.io/docs/faq/#how-to-patch-coredns-and-other-pre-installed-addons

**Comment by stefanprodan** [comment, marked as answer]
For Kubernetes Jobs there is no need to suspend the Flux Kustomization, Flux performs a dry-run and only applies things that change, in case of a Job it will not run it again, unless you change it in Git and you set `Kustomization.spec.force` to `true`. See here how Flux handles Kubernetes Jobs: https://fluxcd.io/docs/use-cases/running-jobs/

**Reply by ntx-ben** [reply]
What I've noticed is that since my `Job` has a `TTL`, once it is removed, Flux will apply it again, and so on forever.

So for now I've removed the `TTL` and it is working as expected.

**Reply by stefanprodan** [reply]
Yeap we should add this to docs, Jobs with TTL are not supported.
